gm4_heart_canisters/data/heart_canisters/functions/check.mcfunction

```diff
@@ -0,0 +1,11 @@
+scoreboard players set @s[nbt={Inventory:[{Count:1b,tag:{gm4_heart_canister_tier:1b}}]}] heart_can_health 1
+scoreboard players set @s[nbt={Inventory:[{Count:2b,tag:{gm4_heart_canister_tier:1b}}]}] heart_can_health 2
+scoreboard players set @s[nbt={Inventory:[{Count:3b,tag:{gm4_heart_canister_tier:1b}}]}] heart_can_health 3
+scoreboard players set @s[nbt={Inventory:[{Count:4b,tag:{gm4_heart_canister_tier:1b}}]}] heart_can_health 4
+scoreboard players set @s[nbt={Inventory:[{Count:5b,tag:{gm4_heart_canister_tier:1b}}]}] heart_can_health 5
+
+scoreboard players add @s[nbt={Inventory:[{Count:1b,tag:{gm4_heart_canister_tier:2b}}]}] heart_can_health 1
```

Comment:
By using add here, a player would be able to use a tier 2 heart can as a regular heart can. The second tier heart cans should only be active if the player also has 5 tier 1's in their inventory.